src/main/java/com/amazon/opendistroforelasticsearch/sql/esdomain/mapping/FieldMapping.java

```diff
@@ -0,0 +1,132 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.esdomain.mapping;
+
+import com.amazon.opendistroforelasticsearch.sql.domain.Field;
+import com.amazon.opendistroforelasticsearch.sql.utils.StringUtils;
+
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static org.elasticsearch.action.admin.indices.mapping.get.GetFieldMappingsResponse.FieldMappingMetaData;
+
+/**
+ * Field mapping that parses native ES mapping.
+ *
+ * NOTE that approaches in this class are NOT reliable because of the ES mapping query API used.
+ * We should deprecate this in future and parse field mapping in more solid way.
+ */
+public class FieldMapping {
+
+    /** Name of the Field to be parsed */
+    private final String fieldName;
+
+    /** Native mapping information returned from ES */
+    private final Map<String, FieldMappingMetaData> typeMappings;
+
+    /** Maps a field name to Field object that specified in query explicitly */
+    private final Map<String, Field> specifiedFieldsByName;
+
+
```

Comment:
nitpick: unnecessary empty line

Reply:
I'm used to adding one more to separate member fields and constructor clearly :) removing.